### DISCUSSION TITLE
Disable NFC in Xdrip?

### DISCUSSION THREAD

**Discussion by atldad** [NightscoutFoundation/xDrip#2398] [Q&A]
Hello!

Recently I enabled NFC to use google pay on my pixel 6a. However, when I did that every-time I opened the xdrip app I would get constant NFC pairing noises. This is because I also keep my credit card in a wallet on the back of my phone and xdrip was finding my credit card (that has NFC as well).

is there a way to disable NFC pairing in the xdrip app or decline that specific permission so it doesn't use NFC when I have the app open?

**Comment by psonnera** [comment]
This is already an available feature. https://xdrip.readthedocs.io/en/latest/install/libreNFC/#enabling-nfc
Disable scan when not in xDrip+

**Comment by atldad** [comment]
Thank you for the response.

I don't see that option in settings, was it added recently?

 I'm running version e13a76e-2022.08.03.

**Reply by atldad** [reply]
Yikes, apologies! I've updated the post.

**Comment by Navid200** [comment]
@atldad Would you please explain what CGM you use?
In other words, what option do you have enabled under Settings &#8722;> Hardware Data Source?

**Comment by atldad** [comment]
dexcom g6

I have g5/g6 transmitter as the hardware source.

**Reply by Navid200** [reply]
This is just a curiosity:

If you look at the system status page, you can see a connection once every 5 minutes.  
When it shows last connected 4 minutes ago, if you wait  a little bit, it will show last connected 5 seconds ago.  And it will keep going up.

The next time that happens, and you let it show last connected, 1 minute ago, if you temporarily change the hardware data source to Libre Bluetooth and then tap on NFC scan features, by any chance, is use NFC feature enabled?
If yes, disable it.

As long as you do this and go back out and change the hardware data source back to G5/G6 in less than 2 minutes, your Dexcom transmitter will never know what you did.  

Do you have any blood glucose meter or pen that you have linked with xDrip?

**Reply by atldad** [reply]
Just tried this - NFC was disabled.

**Comment by psonnera** [comment]
I wanted to write that only Libre sensors enabled NFC, and if your data source is G6 you shouldn't indeed see that menu.
Still, some NFC meters and pens have been added, might be there https://xdrip.readthedocs.io/en/latest/use/glucometer/#nfc-meter or there https://xdrip.readthedocs.io/en/latest/use/pens/#novopen-insulin-pen

**Comment by atldad** [comment]
I do not have any other linked items (just the dexcom sensor).

My issue is - 

Given: 
g6/g5 selected as hardware in the xdrip app
NFC enabled in phone settings

When:
Xdrip App is open and my credit card is next to my phone

Then:
Phone constantly dings that's it's found an NFC device. NOTE - it appears to only do this in the main graph screen. It doesn't do it in the settings screen.

It stops when I do one or more of the following:
1. Move credit card away
2. close the app
3. turn off NFC

Since I'd like to keep my credit card in my wallet attached to my phone and I need to open the app occassionally - is there any way to just prevent the app from searching for NFC items?

**Comment by psonnera** [comment]
@atldad what is the exact message you see on your phone?

**Reply by atldad** [reply]
No error message. Just the "nfc connected" sound/notification going off constantly while the app is open and graph screen displayed.

It does NOT Make the notification when I'm in the settings screen though
